From 22cf4bc3d6dbc00b1db1cd87b77e8fcbb44d6a0c Mon Sep 17 00:00:00 2001
From: Adrian DC <radian.dc@gmail.com>
Date: Tue, 25 Jul 2017 01:40:39 +0200
Subject: [PATCH] sepolicy: Restore support for legacy f_adb interface

 * Restore full support of USB and adb on devices
    using the legacy f_adb kernel drivers

This reverts commit a3bc3cffdfd6abfcdd3cde26eb543619b183efa2.
Change-Id: Iccc8e781b79ebc301c18d9f2e177a1957d2e90c3
Signed-off-by: Adrian DC <radian.dc@gmail.com>
---
 private/adbd.te       | 3 ++-
 private/file_contexts | 1 +
 public/device.te      | 1 +
 public/recovery.te    | 3 ++-
 4 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/private/adbd.te b/private/adbd.te
index 52597ebb..d42f2960 100644
--- a/private/adbd.te
+++ b/private/adbd.te
@@ -23,7 +23,8 @@ allow adbd self:capability setpcap;
 # Create and use network sockets.
 net_domain(adbd)
 
-# Access /dev/usb-ffs/adb/ep0
+# Access /dev/android_adb or /dev/usb-ffs/adb/ep0
+allow adbd adb_device:chr_file rw_file_perms;
 allow adbd functionfs:dir search;
 allow adbd functionfs:file rw_file_perms;
 
diff --git a/private/file_contexts b/private/file_contexts
index 4485b953..0af67c1f 100644
--- a/private/file_contexts
+++ b/private/file_contexts
@@ -65,6 +65,7 @@
 /dev/adf-interface[0-9]*\.[0-9]*	u:object_r:graphics_device:s0
 /dev/adf-overlay-engine[0-9]*\.[0-9]*	u:object_r:graphics_device:s0
 /dev/alarm		u:object_r:alarm_device:s0
+/dev/android_adb.*	u:object_r:adb_device:s0
 /dev/ashmem		u:object_r:ashmem_device:s0
 /dev/audio.*		u:object_r:audio_device:s0
 /dev/binder		u:object_r:binder_device:s0
diff --git a/public/device.te b/public/device.te
index 4a3bec91..9415d836 100644
--- a/public/device.te
+++ b/public/device.te
@@ -1,6 +1,7 @@
 # Device types
 type device, dev_type, fs_type;
 type alarm_device, dev_type, mlstrustedobject;
+type adb_device, dev_type;
 type ashmem_device, dev_type, mlstrustedobject;
 type audio_device, dev_type;
 type audio_timer_device, dev_type;
diff --git a/public/recovery.te b/public/recovery.te
index f55dc8a5..7875adac 100644
--- a/public/recovery.te
+++ b/public/recovery.te
@@ -71,7 +71,8 @@ recovery_only(`
 
   allow recovery kernel:system syslog_read;
 
-  # Access /dev/usb-ffs/adb/ep0
+  # Access /dev/android_adb or /dev/usb-ffs/adb/ep0
+  allow recovery adb_device:chr_file rw_file_perms;
   allow recovery functionfs:dir search;
   allow recovery functionfs:file rw_file_perms;
 
